skt/__init__.py

```diff
@@ -503,24 +508,53 @@ def mktgz(self, clean=True):
 
         logging.info("building kernel: %s", args)
 
-        mk = subprocess.Popen(args,
-                              stdout=subprocess.PIPE,
-                              stderr=subprocess.STDOUT)
-        (stdout, stderr) = mk.communicate()
-        for line in stdout.split("\n"):
-            m = re.match("^Tarball successfully created in (.*)$", line)
-            if m:
-                tgzpath = m.group(1)
-                break
-
-        fpath = None
-        if tgzpath is not None:
-            fpath = "/".join([self.path, tgzpath])
+        with io.open(self.buildlog, 'wb') as writer, \
+                io.open(self.buildlog, 'rb') as reader:
+            make = subprocess.Popen(args,
+                                    stdout=writer,
+                                    stderr=subprocess.STDOUT)
+
+            timer = Timer(timeout, make.terminate)
+            # If the process is cancelled, it doesn't need to wait for anything
+            timer.setDaemon(True)
```

Comment:
Thanks for the comment above, but what does this line actually do? I'm unfamiliar with this API, and wasn't able to find documentation on this quickly.

Reply:
I have rewritten the comment, I hope it's understandable now

Reply:
Sorry, it's probably obvious to regular Python developers (and if that's true, perhaps no comment is necessary), but not to me. What's the difference between `setDaemon(False)` and `setDaemon(True)` in timer behavior? A link to documentation would also be fine. Thanks :)

Reply:
The difference in terms of behavior between a daemon timer and a non-daemon timer is none. Timer is a subclass of Thread, that means it is necessary to read the thread section https://docs.python.org/2/library/threading.html#thread-objects, I have to say that the doc is not clear here... The long story short for our case is, if we don't set as daemon this timer, the main process cannot be stopped (e.g. Ctrl+C or SIGTERM) because it will keep waiting until the timer/thread finish. We can solve this too, calling timer.cancel() on SIGTERM or Ctrl+C but I think this way is much simpler.

Now that you mention it, I think it'd be better remove the comment here...

Reply:
I see, thank you for the explanation :)